kyo-data/shared/src/main/scala/kyo/SafeClassTag.scala

```diff
@@ -0,0 +1,111 @@
+package kyo
+
+import kyo.Maybe
+import kyo.internal.SafeClassTagMacro
+import scala.annotation.tailrec
+import scala.quoted.*
+
+opaque type SafeClassTag[A] >: SafeClassTag.Element = Class[?] | SafeClassTag.Element
+
+object SafeClassTag:
```

Comment:
2 general comments
1. Do you think the naming will confuse users? Seems okay to me, just not fully descriptive... top level Scaladoc?
2. Do we want a `show` method? I don't remember what a Class will be serialized as.

Reply:
The name isn't great but I couldn't find a good alternative. I've pushed scaladocs to explain its purpose + the `show` method.